lib/database.js

```diff
@@ -41,6 +41,13 @@ exports.DatabaseAPI = function DatabaseAPI(hoodie, options) {
 
   db.add = function (type, attrs, callback) {
     attrs.type = type;
+
+    // we can't prepare a couch `id` without a `doc.id`, so
+    // let's check whether it exists first...
+    if (!attrs.id) {
+      return callback(new Error('No `id` associated with this Hoodie object'));
```

Comment:
Maybe make this “Hoodie objects must have an `id` property” to make things a little clearer? (now that my brain is more awake :) — Otherwise top notch patch will merge right away :) Thanks for sticking with me here!